lib/components/ModalWrapper/index.tsx

```diff
@@ -45,29 +47,33 @@ class ModalWrapperImpl extends React.PureComponent<Props, {}> {
    is definitely outside ModalWrapper before calling hide
    **/
   handleClickOutside(e: React.SyntheticEvent<HTMLElement>) {
-    if (!this.modalWrapper) return;
-
     const target = e.target as HTMLElement;
-    if (e.currentTarget === target && this.props.onCancel) this.props.onCancel(e);
+    const isFinalTarget = e.currentTarget === target;
+    if (!isFinalTarget || !this.modalWrapper || !this.props.onClickOutside || ! this.props.onCancel) return;
```

Comment:
Leverage existing `onWrapperCancel`.
Beware, your test should have been : `!isFinalTarget || !this.modalWrapper || !(this.props.onClickOutside || this.props.onCancel)`